src/project.cc

```diff
@@ -572,13 +574,20 @@ std::vector<Project::Entry> LoadCompilationEntriesFromDirectory(
     // Try to load compile_commands.json, but fallback to a project listing.
   } else {
     project->mode = ProjectMode::ExternalCommand;
-#if defined(_WIN32)
-// TODO
-#else
-    char tmpdir[] = "/tmp/cquery-compdb-XXXXXX";
-    if (!mkdtemp(tmpdir))
-      return {};
-    comp_db_dir = tmpdir;
+
+    // generally it would be nice if we could just let clang load the
+    // compilation database (compile_commands.json) from memory.
+    // However, clang insists on reading compile_commands.json from a
+    // directory, so we create a temporary directory just for clang to read
+    // from.
+
+    char templ[] = "/tmp/cquery-compdb-XXXXXX";
+    auto tmpdir = TryMakeTempDirectory(templ);
```

Comment:
Since template string is ignored on windows, I'd suggest just hardcoding it for platform_posix.cc and remove the parameter.

Reply:
The idea was that `TryMakeTempDirectory()` might also be used somewhere else with a different template string.
But either is fine for me.